pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2437,3 +2449,59 @@ func TestServiceRegistryCreateWithoutExistingNetwork(t *testing.T) {
 		t.Fatalf("expected error; got none")
 	}
 }
+
+func TestServiceRegistryCreateWithNetworkOfArktosIPAM(t *testing.T) {
+	origFeatureGate := newFeatureGateWithPerNetworkServiceIPAlloc()
+	defer func() {
+		utilfeature.DefaultFeatureGate = origFeatureGate
```

Comment:
Why we need set utilfeature.DefaultFeatureGate = origFeatureGate when the test case exit?

Reply:
got your point. However, this case is purposed as the most basic scenario for feature of per-network service ip allocation; in case it fails, we know for sure it is broken at this feature, not around multi-tenancy.